pkg/csi_driver/node.go

```diff
@@ -141,6 +141,18 @@ func (s *nodeServer) NodePublishVolume(ctx context.Context, req *csi.NodePublish
 		fuseMountOptions = joinMountOptions(fuseMountOptions, []string{"token-server-identity-provider=" + identityProvider})
 	}
 
+	node, err := s.k8sClients.GetNode(s.driver.config.NodeID)
+	if err != nil {
+		return nil, status.Errorf(codes.NotFound, "failed to get node: %v", err)
+	}
+
+	val, ok := node.Labels[clientset.GkeMetaDataServerKey]
+	// If Workload Identity is not enabled, the key should be missing; the check for "val == false" is just for extra caution
+	isWorkloadIdentityDisabled := val == "false" || !ok
```

Comment:
"false" is not a valid value according to GCW which validates this field: 
```
              allowedKeysWithValues:
              - key: "iam.gke.io/gke-metadata-server-enabled"
                pattern: "true"
```

If you want to keep this extra check, then I think it makes sense to change it to 
isWorkloadIdentityDisabled := val != "true"